src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolverOptionsSetup.cs

```diff
@@ -0,0 +1,101 @@
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryWorkerConfig);
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            _scriptHostManager = scriptHostManager ?? throw new ArgumentNullException(nameof(scriptHostManager));
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        }
+
+        public void Configure(WorkerConfigurationResolverOptions options)
+        {
+            var configuration = GetRequiredConfiguration();
+            options.WorkersDirPath = GetWorkersDirPath(configuration);
+        }
+
+        internal string GetDefaultWorkersDirectory()
+        {
+            var assemblyDir = AppContext.BaseDirectory;
+            string workersDirPath = Path.Combine(assemblyDir, RpcWorkerConstants.DefaultWorkersDirectoryName);
+
+            if (!_fileSystem.Directory.Exists(workersDirPath))
+            {
+                // Site Extension Path. Default to parent directory.
+                var parentDir = _fileSystem.Directory.GetParent(assemblyDir.TrimEnd(Path.DirectorySeparatorChar)).FullName;
+                workersDirPath = _fileSystem.Path.Combine(parentDir, RpcWorkerConstants.DefaultWorkersDirectoryName);
+            }
+            return workersDirPath;
+        }
+
+        private string GetWorkersDirPath(IConfiguration configuration)
+        {
+            var workersDirectorySection = configuration?.GetSection($"{RpcWorkerConstants.LanguageWorkersSectionName}:{WorkerConstants.WorkersDirectorySectionName}");
+
+            if (!string.IsNullOrEmpty(workersDirectorySection?.Value))
+            {
+                return workersDirectorySection.Value;
+            }
+
+            return GetDefaultWorkersDirectory();
+        }
+
+        private IConfiguration GetRequiredConfiguration()
+        {
+            var configuration = _configuration;
+            string requiredSection = $"{RpcWorkerConstants.LanguageWorkersSectionName}:{WorkerConstants.WorkersDirectorySectionName}";
+            _ = GetConfigurationSectionValue(_configuration, nameof(_configuration), requiredSection);
+
+            if (_scriptHostManager is IServiceProvider scriptHostManagerServiceProvider)
+            {
+                var latestConfiguration = scriptHostManagerServiceProvider.GetService<IConfiguration>();
+
+                string value = GetConfigurationSectionValue(latestConfiguration, nameof(latestConfiguration), requiredSection);
+
+                if (!string.IsNullOrEmpty(value))
+                {
+                    var inMemoryCollection = new Dictionary<string, string> { { requiredSection, value } };
+                    configuration = new ConfigurationBuilder()
+                                            .AddConfiguration(_configuration)
+                                            .AddInMemoryCollection(inMemoryCollection)
+                                            .Build();
```

Comment:
I thought we agreed that we don't need a rebuild of configuration based on our last sync since the plan was to check both configurations one by one (latest first).